workbench-core/authorization/src/dynamicAuthorization/dynamoDBDynamicPermissionsPlugin.ts

```diff
@@ -0,0 +1,491 @@
+        .update(key, {
+          item,
+          condition
+        })
+        .execute();
+    } catch (err) {
+      if (err.name === 'ConditionalCheckFailedException') {
+        throw new GroupAlreadyExistsError('Group already exists');
+      }
+      throw err;
+    }
+    return { created: true };
+  }
+
+  public async deleteGroup(deleteGroupRequest: DeleteGroupRequest): Promise<DeleteGroupResponse> {
+    this._isInit();
+
+    // Remove all users from group
+    const { groupId } = deleteGroupRequest;
+    const { userIds } = await this.getUsersFromGroup({ groupId });
+    const removeUsersPromises = _.chunk(userIds, 25).map(async (userIdsChunk) => {
+      const deleteKeys = userIdsChunk.map((userId) => {
+        return {
+          pk: this._composeIdentityKey(this._assignedGroupPrefix, 'USER', userId),
+          sk: this._composeIdentityKey(this._assignedGroupPrefix, 'GROUP', groupId)
+        };
+      });
+      await this._awsService.helpers.ddb
+        .batchEdit({
+          addDeleteRequests: deleteKeys
+        })
+        .execute();
+    });
+    await Promise.all(removeUsersPromises);
+    //Remove all permissions associated to group
+    const { identityPermissions } = await this.getIdentityPermissionsByIdentity({
+      identityType: 'GROUP',
+      identityId: groupId
+    });
+    const removeIdentityPermissionsPromises = _.chunk(identityPermissions, 25).map(
+      async (identityPermissionsChunk) => {
+        const deleteKeys = identityPermissionsChunk.map((identityPermission) => {
+          const item = this._createItemFromIdentityPermission(identityPermission);
+          return {
+            pk: item.pk,
+            sk: item.sk
+          };
+        });
+        await this._awsService.helpers.ddb
+          .batchEdit({
+            addDeleteRequests: deleteKeys
+          })
+          .execute();
+      }
+    );
+    await Promise.all(removeIdentityPermissionsPromises);
+    // Remove group entry
+    const groupKey = {
+      pk: this._composeIdentityKey(this._groupPrefix, 'GROUP', deleteGroupRequest.groupId),
+      sk: this._composeIdentityKey(this._groupPrefix, 'GROUP', deleteGroupRequest.groupId)
+    };
+
+    await this._awsService.helpers.ddb.delete(groupKey).execute();
+    return { deleted: true };
+  }
+
+  public async getUserGroups(getUserGroupsRequest: GetUserGroupsRequest): Promise<GetUserGroupsResponse> {
+    this._isInit();
+
+    const key = {
+      name: 'pk',
+      value: this._composeIdentityKey(this._assignedGroupPrefix, 'USER', getUserGroupsRequest.userId)
+    };
+
+    const groupIds: string[] = [];
+
+    const response: QueryCommandOutput = await this._awsService.helpers.ddb
+      .query({
+        key
+      })
+      .execute();
+
+    if (response && response.Items) {
+      response.Items.forEach((item) => {
+        const sk = _.get(item as { [key: string]: unknown }, 'sk') as string;
+        groupIds.push(this._decomposeKey(sk).id);
+      });
+    }
```

Comment:
This could be 
```
const groupIds = response?.Items?.map((item) => {
  const sk = _.get(item as { [key: string]: unknown }, 'sk') as string;
  return this._decomposeKey(sk).id;
}) ?? [];
```